### PR TITLE
feat(data-table): add row sizes

### LINKED ISSUE #1303: Component - Data Table - Development
## Detailed description

Describe in detail the issue you're having. Is this a feature request (new component, new icon), a bug, or a general issue?

> Is this issue related to a specific component?

> What did you expect to happen? What happened instead? What would you like to see changed?

> What browser are you working in?

> What version of the Carbon Design System are you using?

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?

## Steps to reproduce the issue

1.  Step one
2.  Step two
3.  Step three
4.  etc.

## Additional information

- Screenshots or code
- Notes

## Add labels

Please choose the appropriate label(s) from our existing label list to ensure that your issue is properly categorized. This will help us to better understand and address your issue.

### PR DESCRIPTION
**Pull request by jnm2377**
Closes #1303 

Adds static, compact, short, tall, and small toolbar classes.

**To-do**
- [x] fix tall _sortable_ column headers (alignment is off)

### CODE REVIEW

**Comment by joshblack** [comment]
Hey @jnm2377! 👋 Is this something we're trying to get in for 10.3?

**Comment by jnm2377** [comment]
@joshblack I think so. Tyler was gone last week, so he couldn't review, but it should be ready for review once I can push this last change I made today... I'm having trouble pushing. Not sure if I messed something up after the monorepo change. 💀

**Comment by joshblack** [comment]
@jnm2377 sounds good! And let me know if I can help out with any conflicts, I think doing `git pull upstream master` in your local branch should automatically resolve conflicts but let me know if not!

File: packages/components/src/components/data-table/_data-table-action.scss [hunk 1]
```diff
@@ -29,10 +29,6 @@
     height: $layout-04;
     width: 100%;
     justify-content: flex-end;
-    transform: translate3d(0, 0, 0);
```
Comment (asudoh) [inline packages/components/src/components/data-table/_data-table-action.scss]: Is removal here intentional?

**Comment by designertyler** [comment]
The Default, Compact, Short, Tall, and Zebra have an added 1px border added to the top and bottom of the`th`.

The select, small select, short select, tall select, zebra select, expandable, and pagination  `th` do not have a border top or bottom. Can we do the same (no border top or bottom) on the others as well?

File: packages/components/src/components/data-table/_data-table-action.scss (hunk 1 above)
Comment (jnm2377) [reply packages/components/src/components/data-table/_data-table-action.scss]: Ah, no. Good catch! Thank you!

**Comment by designertyler** [comment]
The "Primary +" text and icon are 3px too low in the button on the small select.

image

When text wraps in the header it should be left aligned and horizontally aligned with the other column title instead of vertically centered. 

Screen Shot 2019-05-15 at 10 14 48 AM

**Comment by designertyler** [comment]
Default, compact, and zebra data table column header text is too high (short looks good). It looks like they're given too little padding above and below which leaves an extra 2 px below.

image

Tall also has some weird padding that leaves the text 1px too low.

**Review by asudoh — COMMENTED** [review]
LGTM from my perspective on the code, thanks @jnm2377! - Wanted to have eyes on @lovemecomputer considering potential impact on her sticky header work.

**Review by lovemecomputer — APPROVED** [review]
looks good!

**Comment by joshblack** [comment]
@jnm2377 any chance you have time to fix the merge conflicts?

**Comment by asudoh** [comment]
@jnm2377 Would you be able to re-run Prettier? Thanks!

**Comment by jnm2377** [comment]
FYI: this will just be focusing on non-sortable size variations. Sortable headers have a bug when there's text overflow (which this PR does not address): https://github.com/carbon-design-system/carbon/issues/2830

**Comment by asudoh** [comment]
@jnm2377 I'm guessing that your above comment is to @designertyler's review comment, is it correct?

**Comment by jnm2377** [comment]
@asudoh yes, I talked to him about clarifying scope for this PR and we both agreed that we'll treat sortable header as a separate issue.

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @jnm2377!

**Pull request merged**